problems.js

```diff
@@ -0,0 +1,40 @@
+// ## Problem1 (https://leetcode.com/problems/construct-binary-tree-from-inorder-and-postorder-traversal/)
+
+var buildTree = function(preorder, inorder) {
+    let hash = {};
+    inorder.forEach((e, i)=>{hash[e] = i});
+    
+    let recur = function(start, end) {
+        if (start > end) return null;
+        let root = new TreeNode(preorder.shift());
```

Comment:
Shift will add O(n) time here. Better to maintain a global variable and move it to next index like class.